tests/backend/test_events.py

```diff
@@ -146,4 +146,29 @@ async def test_events_listen_wait_cancelled(backend_asgi_app, alice_ws):
         listen.rep_done = True
 
 
+@pytest.mark.trio
+async def test_events_close_connection_on_backpressure(
+    backend_asgi_app, alice, bob, monkeypatch, backend_authenticated_ws_factory
+):
+    # The channel has a queue of size 1, meaning it will be filled after a single command
+    monkeypatch.setattr("parsec.backend.client_context.AUTHENTICATED_CLIENT_CHANNEL_SIZE", 1)
+    # We need to manually initialize the websockets using the factory here to make
+    # sure the monkeypatch happens before the memory channel creation and thus
+    # give then a size of 1
+    async with backend_authenticated_ws_factory(backend_asgi_app, alice) as alice_ws:
+        async with backend_authenticated_ws_factory(backend_asgi_app, bob) as bob_ws:
+            await events_subscribe(alice_ws)
+
+            await authenticated_ping(bob_ws, "foo")
+            await authenticated_ping(bob_ws, "foo")
+
+        # Alice should be disconnected at this point and exit the loop
+        with trio.fail_after(1):
```

Comment:
This block is in the wrong place: given it is outside of `async with backend_authenticated_ws_factory(backend_asgi_app, alice) as alice_ws`, it means `alice_ws` has been closed when leaving the async with block (you can try this test without mocking `AUTHENTICATED_CLIENT_CHANNEL_SIZE` to see the issue)

So you do this check within the async with block because we check that the connection has been closed by the server

Reply:
nevermind I misread that ^^